x-pack/platform/plugins/shared/alerting_v2/server/routes/update_esql_rule_route.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import { schema } from '@kbn/config-schema';
+import { getSpaceIdFromPath } from '@kbn/spaces-utils';
+import type { KibanaRequest, KibanaResponseFactory } from '@kbn/core-http-server';
+import type { SavedObjectsServiceStart } from '@kbn/core-saved-objects-server';
+import type { HttpServiceStart } from '@kbn/core-http-server';
+import type { EncryptedSavedObjectsPluginStart } from '@kbn/encrypted-saved-objects-plugin/server';
+import type { TaskManagerStartContract } from '@kbn/task-manager-plugin/server';
+import type { SpacesPluginStart } from '@kbn/spaces-plugin/server';
+import type { SecurityPluginStart } from '@kbn/security-plugin/server';
+import type { Logger as KibanaLogger } from '@kbn/logging';
+import { inject, injectable, optional } from 'inversify';
+import { Logger, PluginStart } from '@kbn/core-di';
+import { CoreStart, Request, Response } from '@kbn/core-di-server';
+import { DEFAULT_ALERTING_V2_ROUTE_SECURITY } from './constants';
+import { ESQL_RULE_SAVED_OBJECT_TYPE } from '../saved_objects';
+import { updateEsqlRule, updateEsqlRuleDataSchema } from '../application/esql_rule/methods/update';
+
+import {
+  createAPIKey,
+  getAuthenticationAPIKey,
+  isAuthenticationTypeAPIKey,
+} from '../application/esql_rule/lib/api_key';
+
+const INTERNAL_ESQL_RULE_API_PATH = '/internal/alerting/esql_rule';
+
+const updateEsqlRuleParamsSchema = schema.object({
+  id: schema.string(),
+});
+
+@injectable()
+export class UpdateEsqlRuleRoute {
+  static method = 'put' as const;
```

Comment:
```suggestion
  static method = 'patch' as const;
```

As we want to allow partial updates and not a full replacement.